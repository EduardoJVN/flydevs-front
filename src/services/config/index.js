export const route = process.env.REACT_APP_API_URL