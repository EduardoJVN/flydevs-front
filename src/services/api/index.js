import axios from "axios";
import {route} from "../config"
//traer listado de pokemons 20 (sin capturar)
export const getPokemons = async () =>{
    const data = axios.get(`${route}pokemons/`).then((res)=>{
        return {message:res.data,error:false}
    }).catch(err=>{
        return {message:err.response.data.message,error:true}
    })
    return data
}


//busqueda de pokemon
export const searchPokemons = async (text) =>{
    const data = axios.get(`${route}pokemon/${text}`).then((res)=>{
        return {message:res.data,error:false}
    }).catch(err=>{
        return {message:err.response.data.message,error:true}
    })
    return data
}