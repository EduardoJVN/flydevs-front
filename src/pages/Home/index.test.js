import React from 'react';
import App from './index';
import { act } from 'react-dom/test-utils';
import Enzyme , {mount, render} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";
import ReactDOM from 'react-dom';
import {searchPokemons,getPokemons} from '../../services/api'
Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../services/api')

it('renders App correctly', async () => {

    const pokemon = 'ditto'
    let response = []
    response.push({name:pokemon,image:"test"})
    searchPokemons.mockReturnValue(Promise.resolve({message:response,error:false}))
    getPokemons.mockReturnValue(Promise.resolve({message:response,error:false}))
    await act( async () => {

        const wrapper = await mount(<App/>);

        expect(wrapper.find('.image').length).toBe(0)

        expect(wrapper.find('input').at(0).prop('value')).toBe("")

        wrapper.find('input').at(0).simulate('change', { target: { value: pokemon } })

        expect(wrapper.find('input').at(0).prop('value')).toBe(pokemon)
        
        wrapper.find('button').simulate('click')

        await Promise.resolve();

        wrapper.update()

        expect(wrapper.find('.image').length).toBe(1)

        var el = document.createElement( 'div' );
        el.innerHTML = wrapper.html()
        expect(el).toMatchSnapshot();
        })    
});