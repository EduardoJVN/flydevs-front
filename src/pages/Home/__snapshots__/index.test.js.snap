// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-root-4"
    />
    <main
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <h1
        class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        data-testid="title"
      >
        Pokemon finder
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        El que quiere pokemons, que los busque.
      </h6>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
        >
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-input-2"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              Nombre del pokemon
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                type="text"
                value="ditto"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Buscar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h1
            class="MuiTypography-root tableTitle MuiTypography-h6 MuiTypography-gutterBottom"
          >
            Resultados de la b√∫squeda
          </h1>
        </div>
        <div
          class="MuiGrid-root itemList MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-sm-4 MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2"
              >
                <img
                  class="image"
                  src="test"
                  style="width: 100%; height: 150px;"
                />
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-md-10 MuiGrid-grid-lg-10 MuiGrid-grid-xl-10"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5"
                >
                  ditto
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
